libtsuba/include/tsuba/ParquetReader.h

```diff
@@ -17,22 +23,67 @@ class KATANA_EXPORT ParquetReader {
     int64_t length;
   };
 
-  /// \returns a Reader that will read a table from storage location optionally
-  /// reading only part of the table defined by \param slice
+  struct ReadOpts {
+    /// if true (default) make sure canonical types are used and table columns
+    /// are not chunked
+    bool make_safe{true};
+
+    /// if provided, slice the resulting table so that it only contains
+    /// Slice.length rows starting from Slice.offset
+    std::optional<Slice> slice{std::nullopt};
+
+    static ReadOpts Defaults() { return ReadOpts{}; }
+  };
+
+  /// build a reader that will read a table from storage location optionally
+  /// reading only part of the table.
+  /// \param opts an opt structure detailing how reads should behave (see
+  ///    the ReadOpts struct definition for details)
   static katana::Result<std::unique_ptr<ParquetReader>> Make(
-      std::optional<Slice> slice = std::nullopt);
+      ReadOpts opts = ReadOpts::Defaults());
 
-  /// read table from \param uri
-  katana::Result<std::shared_ptr<arrow::Table>> ReadFromUri(
+  /// read table from storage
+  ///   \param uri an identifier for a parquet file
+  katana::Result<std::shared_ptr<arrow::Table>> ReadTable(
       const katana::Uri& uri);
 
+  /// read part of a table from storage
+  /// n.b. support for the `slice` read option is missing here
+  ///   \param uri an identifier for a parquet file
+  ///   \param column_bitmap must have the same length as the number of columns
+  ///      in the table in the parquet file. The loaded table will only contain
+  ///      columns at indexes that are true in the bitmap
+  katana::Result<std::shared_ptr<arrow::Table>> ReadTable(
+      const katana::Uri& uri, const std::vector<bool>& column_bitmap);
```

Comment:
wondering: `ReadTable` takes a bitmap while `ReadColumn` takes a column index. It might be more uniform to callers for this to take a list of indices as well. I suppose this raises the question about what we would do if there were duplicate indices or how permutations would be handled, though the underlying `arrow::Table` could model those aspects as well if we wanted.